packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -353,11 +353,25 @@ export const Table = <
 
   // Original type for some reason is missing sub-columns
   type ColumnType = Column<T> & {
-    columns: Column<T>[];
+    columns: ColumnType[];
   };
-  const hasManualSelectionColumn = (columns as ColumnType[])[0]?.columns.some(
-    (column) => column.id === SELECTION_CELL_ID,
+  const flattenColumns = React.useCallback(
```

Comment:
Maybe move out this and `type` from `Table` component above?

Reply:
Moved